### PR TITLE
Feature/profile classe teacher student

### PR DESCRIPTION
**Pull request by giogio21**

### CODE REVIEW

**Comment by arnauds5** [comment]
It seems that a lot of code is dupplicated here.
If the only differences are not having a sciper for teacher, and to differentiate between student/teacher, maybe it could be done by : 
- Having a nullable sciper
- Adding an attribute with an enum of Student/Teacher/etc

File: app/src/main/java/com/sdp13epfl2021/projmag/model/ImmutableProfile.kt [hunk 1]
```diff
@@ -21,18 +21,19 @@ class ImmutableProfile private constructor(val lastName : String, val firstName
...
         fun build(lastName : String, firstName : String,
-                  age : Int, gender : Gender, sciper : Int, phoneNumber :String) : Result<ImmutableProfile> {
+                  age : Int, gender : Gender, sciper : Int?, phoneNumber :String, role : Role) : Result<ImmutableProfile> {
             return when{
                 lastName.length > MAX_LAST_NAME_SIZE ->  Failure("last name is more than $MAX_LAST_NAME_SIZE characters")
                 firstName.length > MAX_FIRST_NAME_SIZE -> Failure("first name is more than $MAX_FIRST_NAME_SIZE characters")
                 age > MAX_AGE -> Failure("age is more than $MAX_AGE")
                 age <  MIN_AGE -> Failure("age is less than $MIN_AGE")
-                sciper < 0 -> Failure("sciper can't be a negative value")
+                (sciper!! < 0) -> Failure("sciper can't be a negative value")
```
Comment (arnauds5) [inline app/src/main/java/com/sdp13epfl2021/projmag/model/ImmutableProfile.kt]: If we allow null sciper, this check will throw a NullPointerException for null value
```suggestion
                (sciper != null && sciper < 0) -> Failure("sciper can't be a negative value")
```

**Review by arnauds5 — APPROVED** [review]
Looks good to me.
Maybe adding the video part, only in the other PR would have been better, but seems ok.

**Comment by PaulCoral** [comment]
Same as in #82 

- [1](https://github.com/asfoury/projmag/pull/82#discussion_r614090406)
- [2](https://github.com/asfoury/projmag/pull/82#discussion_r614093646)

File: app/src/main/java/com/sdp13epfl2021/projmag/model/Role.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        }
+           },TEACHER{
+        override fun toString(): String {
+            return "Teacher"
+        }
+    },OTHER{
+        override fun toString(): String {
+            return "Other"
+        }
+    }
```
Comment (PaulCoral) [inline app/src/main/java/com/sdp13epfl2021/projmag/model/Role.kt:16]: [Suggestion] this is a shorter and clearer syntax, but also ok to keep it
```kotlin
override fun toString(): String = "Other"
```

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -13,9 +22,15 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Projmag">
+
+        <activity android:name=".VideoCv">
+
+        </activity>
+
```
Comment (PaulCoral) [inline app/src/main/AndroidManifest.xml]: **There's no VideoCv Activity**

**Review by PaulCoral — APPROVED** [review]
Looks good to me
**It would be better to remove `.VideoCv` from `AndroidManifest`**

**Pull request merged**